frontend/src/lib/components/UnitPicker/CustomUnitModal.tsx

```diff
@@ -0,0 +1,91 @@
+import { FilterType } from '~/types'
+import { RefCallback, useEffect, useState } from 'react'
+import { LemonModal } from 'lib/components/LemonModal'
+import { LemonButton } from 'lib/components/LemonButton'
+import { PureField } from 'lib/forms/Field'
+import { capitalizeFirstLetter } from 'lib/utils'
+import { LemonInput } from 'lib/components/LemonInput/LemonInput'
+import { HandleUnitChange } from 'lib/components/UnitPicker/UnitPicker'
+
+function chooseFormativeElementValue(
+    formativeElement: 'prefix' | 'postfix' | null,
+    filters: Partial<FilterType>
+): string {
+    if (formativeElement === 'prefix') {
+        return filters.aggregation_axis_prefix || ''
+    }
+
+    if (formativeElement === 'postfix') {
+        return filters.aggregation_axis_postfix || ''
+    }
+
+    return ''
+}
+
+export function CustomUnitModal({
+    isOpen,
+    onSave,
+    formativeElement,
+    filters,
+    onClose,
+    overlayRef,
+}: {
+    isOpen: boolean
+    onSave: (hx: HandleUnitChange) => void
+    formativeElement: 'prefix' | 'postfix' | null
+    filters: Partial<FilterType>
+    onClose: () => void
+    overlayRef: RefCallback<HTMLDivElement>
+}): JSX.Element | null {
+    const [localFormativeElementValue, setLocalFormativeElementValue] = useState<string>(
+        chooseFormativeElementValue(formativeElement, filters)
+    )
+    useEffect(() => {
+        setLocalFormativeElementValue(chooseFormativeElementValue(formativeElement, filters))
+    }, [formativeElement])
+
+    if (formativeElement === null) {
+        return null
+    }
+
+    return (
+        <LemonModal
+            overlayRef={overlayRef}
+            isOpen={isOpen}
+            onClose={onClose}
+            bringToFront={true}
+            title={`Custom ${formativeElement}`}
+            footer={
+                <>
+                    <div className="flex-1">
```

Comment:
Hmm, I think "Cancel" should be right next to "Apply", so `flex-1` isn't needed (example: "New dashboard" modal)

Reply:
A quick search of the code shows we've got multiple ways of adding these buttons... Feels like we could do with an "add a custom footer" or "what's your yes and no options"

Reply:
Yeah, this isn't totally clear, we could really just borrow Ant's `okText`/`onOk` API